core/src/main/java/org/eclipse/hono/util/MessageTap.java

```diff
@@ -25,7 +23,7 @@
  */
 public final class MessageTap implements Consumer<Message> {
 
-    private Consumer<Message> tapConsumer;
+    private final Consumer<Message> tapConsumer;
```

Comment:
Given that
- the class is `final` already
- the member is `private`
why make it additionally `final` itself?
Not that it is wrong or particularly bad, but IMHO this should be superflous, shouldn't it?

Reply:
`final` on the member makes it `const`, where `final` on the class only prevents any derived classes. So what I do here, lets the compiler check if the member gets re-assign and prevents that.